build/Targets/Packages.props

```diff
@@ -258,6 +262,17 @@
     <xunitrunnerconsoleVersion>2.4.1-pre.build.4059</xunitrunnerconsoleVersion>
     <xunitrunnerwpfVersion>1.0.51</xunitrunnerwpfVersion>
     <xunitrunnervisualstudioVersion>2.4.1-pre.build.4059</xunitrunnervisualstudioVersion>
+
+    <!--
+      The CodeStyle layer depends on a prior stable release of the CodeAnalysis assemblies so the NuGet packages
+      produced by the build can be installed in projects building with those older releases.
+    -->
```

Comment:
It seems the property definitions here are violating some principles - this file should only define language agnostic and layer agnostic properties (it should not have any knowledge of the version requirements of any specific layer, such as CodeStyleLayer's required version). I would request @dotnet/roslyn-infrastructure to take a closer look at changes in this file.

Reply:
Keep in mind the primary purpose of this property is to support loading the workspaces layer during compilation, so the entire feature intentionally undermines the layering. When the formatter is moved to the code style layer, we can remove the workspace dependency and these properties will go away at the same time.

Reply:
This feels fairly hacky.  We should be tracking the removal of this with a bug link. 

---
In reply to: [222536172](https://github.com/dotnet/roslyn/pull/28873#discussion_r222536172) [](ancestors = 222536172)

Reply:
➡️ Filed https://github.com/dotnet/roslyn/issues/30472